### PR TITLE
Add ALS to sparkbench/ml

### PR DESCRIPTION
**Pull request by ynXiang**
This PR adds Alternating Least Squares (als) to sparkbench of Hibench. It contains a data-generator to generate RDD[Rating] data, and an ALS algorithm using there data on a product recommendation system.

### CODE REVIEW

**Comment by VinceShieh** [comment]
@ynXiang please add a brief description for this PR

**Comment by VinceShieh** [comment]
@ynXiang Could u pls also upate the README/wiki for the ML workload changes?

**Comment by carsonwang** [comment]
@VinceShieh @mpjlu can you please review this ?

File: conf/workloads/ml/als.conf [hunk 1]
```diff
@@ -0,0 +1,39 @@
+hibench.als.tiny.users          	100
+hibench.als.tiny.products          	100
+hibench.als.tiny.sparsity   		0
+hibench.als.tiny.implicitprefs	       	true
+hibench.als.small.users        		1000
```
Comment (VinceShieh) [inline conf/workloads/ml/als.conf]: remove extra spaces?

File: conf/hibench.conf [hunk 2]
```diff
@@ -3,10 +3,10 @@
 hibench.scale.profile                 tiny
 
 # Mapper number in hadoop, partition number in Spark
-hibench.default.map.parallelism         8
+hibench.default.map.parallelism         138
```
Comment (VinceShieh) [inline conf/hibench.conf]: set this value to default, and doc the value to inform user to set the value properly

File: conf/hibench.conf [hunk 3]
```diff
@@ -3,10 +3,10 @@
...
 
 # Reducer nubmer in hadoop, shuffle partition number in Spark
-hibench.default.shuffle.parallelism     8
+hibench.default.shuffle.parallelism     138
```
Comment (VinceShieh) [inline conf/hibench.conf]: same as above

**Comment by ynXiang** [comment]
@VinceShieh  Problems fixed.

**Comment by VinceShieh** [comment]
LGTM
thanks

**Comment by carsonwang** [comment]
Please remember fix the run_all script for any new added workload.
can you also add this in travis to test it automatically.

**Comment by carsonwang** [comment]
Thanks. Merged.

**Pull request merged**